src/main/java/de/internetsicherheit/brl/bloxberg/cache/ethereum/BloxbergClient.java

```diff
@@ -52,15 +53,43 @@ private EthBlock getEthBlock(BigInteger block) throws IOException {
         return web3j.ethGetBlockByNumber(DefaultBlockParameter.valueOf(block), true).send();
     }
 
-    public BlockWithData getBlockWithData(BigInteger block) throws IOException {
-        this.rawBlock = this.getEthBlock(block);
-        this.blockWithData = new BlockWithData(this.rawBlock.getBlock());
+    private Block getBlockWithData(BigInteger block) throws IOException {
+        EthBlock rawBlock = this.getEthBlock(block);
+        this.blockWithData = new Block(rawBlock.getBlock());
         return this.blockWithData;
     }
 
-    public BigInteger getBlockTimestamp(BigInteger block) throws IOException {
-        BlockWithData blockForTimestamp = this.getBlockWithData(block);
-        return blockForTimestamp.getTimestamp();
+    private boolean isSmartContract(String address) throws IOException {
+
+        EthGetCode addressCodeRequest = web3j.ethGetCode(address, DefaultBlockParameter.valueOf("latest")).send();
+        return !addressCodeRequest.getResult().equals("0x");
+    }
+
+
+    private TransactionAddress.type getTransactionAddressType(TransactionAddress address) throws IOException {
```

Comment:
this is not a `getter` method. Could be renamed to `determineType` or `mapToType`.